streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -389,6 +395,10 @@ private void shutdown() {
             log.error("{} Failed to close restore consumer: ", logPrefix, e);
         }
 
+        // hotfix to improve ZK behavior als long as KAFKA-4060 is not fixed (c.f. KAFKA-4369)
+        // when removing this, make StreamPartitionAssignor#internalTopicManager "private" again
+        partitionAssignor.internalTopicManager.zkClient.close();
```

Comment:
It's not really related to this PR. But as reported in https://issues.apache.org/jira/browse/KAFKA-4369 right now we gat a lot of this messages in the log:
```
[2016-07-14 09:49:21,790] WARN [main-SendThread(127.0.0.1:61385)] Session 0x155e860e62b0007 for server null, unexpected error, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:361)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1022)
```

As we want to remove ZK dependency, nobody picked up this bug and it's super annoying if you run test locally in a loop. We get rid of those log messages with this fix. It's not a completely nice fix but it does the trick -- and I am not sure how long it will take to remove ZK dependency. Thus I though it might be worth to include it for now. Will get removed when ZK dependency gets removed.